packages/qb/pqb/src/queryMethods/then.ts

```diff
@@ -341,32 +342,30 @@ const then = async (
         // afterCommitHooks are executed later after transaction commit,
         // or, if we don't have transaction, they are executed intentionally after other after hooks
         if (afterCommitHooks) {
-          if (
-            trx &&
-            // when inside test transactions, push to a transaction only unless it's the outer user transaction.
-            (!trx.testTransactionCount ||
-              trx.transactionId + 1 > trx.testTransactionCount)
-          ) {
-            (trx.afterCommit ??= []).push(
-              result as unknown[],
-              q,
-              afterCommitHooks,
-            );
+          const boundHooks = afterCommitHooks.map((hook) => {
+            const boundHook = hook.bind(null, result, q);
+            if (hook.name) {
+              Object.defineProperty(boundHook, 'name', { value: hook.name });
+            }
+            return boundHook;
+          });
+
+          if (trx && isInUserTransaction(trx)) {
+            (trx.afterCommit ??= []).push(...boundHooks);
+            trx.catchAfterCommitError = q.q.catchAfterCommitError;
```

Comment:
I possibly tend to agree, but this is replicating the current logic in `main`, which is simply taking the last `q.q.catchAfterCommitError`:

https://github.com/romeerez/orchid-orm/blob/39619d7439edb0492631c681792d0cae8cd1c0e1/packages/qb/pqb/src/queryMethods/transaction.ts#L401-L407

I figured I'd rather have this backwards compatible; calling multiple catchers is an improve that can/should be added separately.

Reply:
Okay, so if that's a current logic, you can leave it as it is and I'll take a note to change it later.
Because in any case, registering callbacks and then not getting them being called isn't good.

Reply:
On second thought, if introducing backwards-incompatible change of removing inner await on hooks, this should better go in a single batch.